add more aether block convert recipe support on jei

baguchi opened the pull request:
This pull request make more jei compat for aether

What Add it on Jei Recipe Support:

Swet Ball Convation 
Ambrosium Convation
![2023-02-19_21 31 25](https://user-images.githubusercontent.com/88538210/219948062-a1a785ce-9cd5-4f66-907e-4c019e74304f.png)


---

I agree to the Contributor License Agreement (CLA).

baguchi committed:
add more aether block convert recipe support on jei

baguchi committed:
now better... I guess

baguchi committed:
now better... I guess

baguchi committed:
deleted IceStone Freezable Recipe support because liquid not render

OzPayn committed:
large scale rebalances

-Zephyr aggression is way lower
-Zephyr spawn rates have been reduced
-Cockatrice spawn rates have been reduced so night time is no longer a nightmare
-Many spawn rates have been mildly adjusted to make useful mobs easier to find
-fiddled with creatureGenerationProbability without really knowing what it does but seems to populate islands a bit better with creatures

baguchi committed:
Merge branch '1.19' into jei_recipe

Commit:
refactor: Subtitles use aether namespace and crafters have unique subtitles

Commit:
Merge branch '1.19' into feat/bconlon/crafting-sounds

Jonathing commented on src/main/java/com/gildedgames/aether/data/generators/AetherLanguageData.java:
```diff
@@ -622,6 +622,8 @@ protected void addTranslations() {
         addGuiText("jei.freezing", "Freezing");
         addGuiText("jei.incubating", "Incubating");
         addGuiText("jei.fuel", "Aether Fuel");
+        addGuiText("jei.swet_ball_convert", "Swet Ball Convert");
+        addGuiText("jei.ambrosium_convert", "Ambrosium Convert");
```
English isn't your mother tongue, which is perfectly fine. I still want to be picky about it though for the sake of consistency, the PR, and also to help you practice.

Instead of "Ambrosium Convert", "Ambrosium Conversion" would be better, since the "-sion" part of the word implies an effect, which is what these JEI recipes explain. Same for "Swet Ball Convert".

Jonathing reviewed (commented):
Good PR. Short and simple. Please do take note about my comment on the English translation keys for swet ball conversion and ambrosium conversion.

Both swet ball conversion and ambrosium conversion are so simple, that their GUI classes for each look very similar (or maybe even are the same). It might be prudent to make a superclass or super interface to reduce the boilerplate when adding more JEI recipes like these. For example, interface `ConversionRecipe` could be interfaced in both `AmbrosiumRecipe` and `SwetBallRecipe` to alleviate this potential problem. Though, that would probably be a job for @bconlon1 or a PR from myself if it came down to it.

Burning-Cactus committed:
fix: Sync eternal day on respawn (#990)

reviewed by bconlon1

merged from #990

Drullkus committed:
fix: Moa render crashing if a Moa is rendered before its type is set (#992)

reviewed by bconlon1

merged from #992

Commit:
fix(entity): Invisbility no longer confuses Slider

bconlon1 committed:
improv: Ambrosium as furnace fuel

OzPayn committed:
Update Zephyr.java

adjusted follow and target ranges

OzPayn committed:
fixed a typo

40 x 40 makes more sense than 40 x 440

baguchan commented:
Thanks for kindly help :)
yeah. It's true that the internals are similar, so it might be possible to omit it.

OzPayn committed:
Mob rebalancing (Deluxe edition) (#997)

Zephyr behaviour has been reworked to be less aggressive
Mob spawning has been balanced to account for nighttime
Zephyrs are also generally less common

OzPayn committed:
adjusted bronze dungeon values

baguchi committed:
oops.... naming fix

OzPayn committed:
Merge branch '1.19' into feat/oz/bronze-dungeon-overlap

bconlon1 committed:
fix(block): Fix another Enchanted Gravitite duplication bug (#994)

reviewed by devilbringer

merged from #994

bconlon1 committed:
fix: Don't drop natural gravitite ore at max y level

OzPayn committed:
more dungeon adjustments

bronze dungeons should overlap less frequently

floating dungeons are more spaced out

OzPayn committed:
improv: Ambrosium as furnace fuel (#996)

OzPayn committed:
fix(entity): Invisbility no longer confuses Slider (#993)

OzPayn committed:
Dungeon Spawning Rebalancing (#998)

bconlon1 committed:
fix: Don't drop natural Gravitite Ore

bconlon1 committed:
fix: Check harvestibility

bconlon1 committed:
fix(entity): Mount and Moa fixes (#995)

closes #981

reviewed by OscarPayn

merged from #995

Commit:
refactor: Crafters actually use the new subtitles

Commit:
Merge branch '1.19' into feat/bconlon/crafting-sounds

Commit:
fix: Implement missing lang entries

OzPayn committed:
refactor(entity): Sun Spirit rebalances (#1000)

- speed has been reduced by .05
- burning bounding box has been extended down and contracted inwards to better reflect the particle visual
- setSecondsOnFire set to 5 from 15 to feel more consistent with other fire sources in the fight
- IceCrystal speed upped to 0.4
- IceCrystal hit speed increased to feel punchier/more accurate

reviewed by bconlon1

merged from #1000

OzPayn committed:
refactor: vignette transparency added (#1001)

changed the inebriation and healing vignette textures to be transparent for better clarity/softer effect

reviewed by bconlon1

merged from #1001

OzPayn committed:
fix: Gravitite Ore drop fixes (#999)

Commit:
refactor: Plant patches have the ability to spawn under overhangs

OzPayn committed:
refactor: Subtitles use aether namespace and crafters have unique subtitles (#991)

OzPayn committed:
refactor: Plant patches have the ability to spawn under overhangs (#1002)

bconlon1 committed:
update: Curios 5.1.2.0 -> 5.1.3.1 (#1003)

Co-authored-by: bconlon <bconlongit@gmail.com>

Burning-Cactus committed:
fix: Remove orange moas (#1004)

Orange moas do not fit in with Aether 1 in its current state. They fit
more in an addon mod.

If the game tries to load an already existing orange moa, there will be
an error in the log. This'll only happen once per moa, so it should be
fine.

Burning-Cactus committed:
Document AetherLevelData

Burning-Cactus committed:
Structure piece comments

Burning-Cactus committed:
build: Add Debug Utils as a runtime dependency (#1005)

https://www.curseforge.com/minecraft/mc-mods/debug-utils-forge

reviewed by bconlon1

merged from #1005

bconlon1 committed:
chore: Reset config.yml cache

Burning-Cactus committed:
Merge large aercloud pieces

Commit:
feat: Experimental lightmap adjustments (commented out)

bconlon1 committed:
ci: Parallel threads

bconlon1 committed:
ci: Cache v5

bconlon1 committed:
ci: Update machine image

bconlon1 committed:
ci: Remove thread count

Burning-Cactus committed:
Failed lighting experiment

bconlon1 committed:
ci: Cache v6

bconlon1 committed:
fix: Book of Lore resets pages properly and lore is more accurate (#943)

reviewed by OscarPayn

merged from #943

bconlon1 committed:
fix: Parachute can no longer auto-deploy when player already has a vehicle

OzPayn committed:
fix: Parachute can no longer auto-deploy when player already has a vehicle (#1008)

closes #1007

bconlon1 committed:
fix: Safer getting of MoaType (#1009)

bconlon1 committed:
build(version): 0.3.1 -> 0.4.0

OzPayn committed:
feat: advancement overhaul (#1013)

Completely new advancement tree!

New root advancement gives correct tab title and icon

Advancements now progress properly through the mod's content

Bunch of new hidden challenges

reviewed by bconlon1

merged from #1013

OzPayn committed:
improved aercloud aesthetics (#1016)

increased opacity on new textures
added shadow casting
added ambient occlusion

Aerclouds now closer resemble the rendering in the original mod, higher
opacity hides opposite faces better and shadows create a softer, less
glassy look.

1.2

![image](https://user-images.githubusercontent.com/6759864/221643241-0c224120-1dca-455d-8f52-86c55354c0a8.png)

this PR

![image](https://user-images.githubusercontent.com/6759864/221643309-b0c0263d-8c89-48e7-af7c-2a7e0b70ceb5.png)

reviewed by bconlon1

merged from #1016

bconlon1 committed:
refactor: Begun implementing new obsidian armor conversion system

Commit:
improv(item): Better Obsidian Armor conversion

Commit:
refactor: White-balanced lightmap

Commit:
fix: Lance can receive more enchantments

Commit:
feat: Green sunset config

OzPayn committed:
fix(item): Lance can receive more enchantments (#1018)

OzPayn committed:
feat: Green sunset config (#1019)

![2023-02-27_17 50
14](https://user-images.githubusercontent.com/67203206/221732172-a0a40b76-960e-411f-afde-e6065a206d8d.png)

Commit:
feat: Config for lightmap

OzPayn committed:
improv(item): Better Obsidian Armor conversion (#1017)

Commit:
Merge branch '1.19' into feat/bconlon/sky-color

OzPayn committed:
feat: Adjust lightmap to not be warm-tinted (#1006)

Now:
![2023-02-27_16 33
41](https://user-images.githubusercontent.com/67203206/221722365-c2292e79-3bd1-47de-b2ae-ebfea002c00f.png)

Previous: 
![2023-02-27_16 33
49](https://user-images.githubusercontent.com/67203206/221722368-1b01528e-4a73-417d-8c9e-d4bea2193a90.png)

bconlon1 committed:
feat: More blue cape recipes

bconlon1 committed:
fix(item): Check for array size in Cloud Staff (#1014)

closes #1011

reviewed by Burning-Cactus

merged from #1014

OzPayn committed:
feat: More blue cape recipes (#1020)

OzPayn committed:
Sun Spirit adjustments

changed speed, damage and fire time values across the Sun Spirit fight

Burning-Cactus committed:
feat: Use entity processing to make bosses for dungeons (#1012)

This fixes a mod conflict with Distant Horizons and gives data packers a new tool to play with

reviewed by OscarPayn

merged from #1012

Burning-Cactus committed:
Merge

bconlon1 committed:
fix(entity): Ensure Moas have a MoaType with /summon (#1015)

Co-authored-by: bconlon <bconlongit@gmail.com>

OzPayn committed:
refactor: Sentry adjustments (#1021)

adjusted max health, movement speed and explosion knockback values

reviewed by bconlon1

merged from #1021

bconlon1 committed:
refactor: Change how zanite accessory durability works (#1024)

reviewed by Burning-Cactus

merged from #1024

bconlon1 committed:
fix: Valkyrie Queen should no longer despawn (#1025)

reviewed by Burning-Cactus

merged from #1025

bconlon1 committed:
refactor: Better particle handling for crystal projectiles (#1026)

reviewed by Burning-Cactus

merged from #1026

bconlon1 committed:
fix: Aerbunny gets properly dismounted under more circumstances (#1027)

closes #1023

reviewed by Burning-Cactus

merged from #1027

bconlon1 committed:
fix: Sun Spirit will ignore invisibility now (#1028)

Co-authored-by: bconlon <bconlongit@gmail.com>

Burning-Cactus committed:
Add isOutOfBounds method to VerticalGradientProcessor for organization

Burning-Cactus committed:
Merge branch '1.19' into feat/burning/world-docs

Burning-Cactus committed:
Silver dungeon data markers

Burning-Cactus committed:
SilverDungeonBuilder

Burning-Cactus committed:
Bronze dungeon clean up

bconlon1 committed:
fix: Moas shouldn't age up without being fed Aechor Petals (#1030)

Co-authored-by: bconlon <bconlongit@gmail.com>

Burning-Cactus committed:
Slider ignores line of sight

OzPayn committed:
rebalanced fuel and timing for utility blocks

Altar repair times shortened
Altar enchanting times shortened
Ambrosium fuel time reduced slightly
Incubation time cut drastically
Freezer timings mildly rebalanced

OzPayn committed:
fix: Slider ignores line of sight (#1034)

OzPayn committed:
fix: rebalanced fuel and timing for utility blocks (#1035)

Altar repair times shortened
Altar enchanting times shortened
Ambrosium fuel time reduced slightly
Incubation time cut drastically
Freezer timings mildly rebalanced

Burning-Cactus committed:
fix: Treasure chest lore, cloud crystal death message, and cold parachute protip are all fixed (#1032)

OzPayn committed:
fix: SunSpirit, FireMinion, and FireCrystal values

rebalanced the sun spirit fight to be more dynamic and mechanically interesting
fire crystals are now much more destructive and require the player to hit them to slow them down
everything hits a lot harder requiring consistent healing and protection
crystal spawning increases dramatically as the Sun Spirit takes damage

Burning-Cactus committed:
Switch aechor plants to the creature category

Burning-Cactus committed:
Rename DungeonTracker to BossRoomTracker

Burning-Cactus committed:
Oops

OzPayn committed:
hopefully final Sun Spirit adjustments (#1022)

changed speed, damage and fire time values across the Sun Spirit fight

fire crystals now slow down if hit

ice crystals are now easier to chase but travel faster when hit

sun spirit travels slower to allow easier aiming and more reliably
dodging

attack and burning values have been adjusted across the board

OzPayn committed:
feat: added swet ball recipes for remaining slimeball recipes (#1038)

added swet slime block and swet sticky piston recipes

OzPayn committed:
fix: added b1.7 zanite and gravitite block textures

moved the 1.2 textures from the base pack into the 1.2 pack and added recreated b1.7 textures to the b1.7 pack

OzPayn committed:
fix: added b1.7 zanite and gravitite block textures (#1039)

moved the 1.2 textures from the base pack into the 1.2 pack and added
recreated b1.7 textures to the b1.7 pack


![image](https://user-images.githubusercontent.com/6759864/223781383-916bf977-55fd-4940-8ab2-6abce5e1068c.png)
beta 1.7 resources

bconlon1 committed:
refactor: Begin setup for adjusting sunset colors (#1029)

Co-authored-by: bconlon <bconlongit@gmail.com>
Co-authored-by: Oscar Payn <oscar@gildedgames.com>

Burning-Cactus committed:
Aechor plant extermination

OzPayn committed:
enhancement: reworked bronze dungeon chest loot

full rework of dungeon chest and boss loot pools
decreased the spawn rate of chest mimics in bronze dungeons

Burning-Cactus committed:
The sun spirit no longer despawns on peaceful

OzPayn committed:
feat: Change aechor plant spawn conditions to have a random check (#1036)

Hopefully this will help to control their spawn rates.

OzPayn committed:
refactor: Structure organization and documentation (#1031)

OzPayn committed:
fix: The sun spirit no longer despawns on peaceful (#1041)

OzPayn committed:
refactor: BossMob cleanup (#1037)

Burning-Cactus committed:
Phoenix arrows won't burn endermen

OzPayn committed:
fix: Phoenix arrows don't burn endermen (#1042)

bconlon1 committed:
chore: Fix Twitter social badge

Thanks Elon.

[ci skip]

bconlon1 committed:
chore: Fix Modrinth download badges

[ci skip]

OzPayn committed:
enhancement: silver dungeon loot rebalance

full rework of silver dungeon chest and boss loot pools
decreased the spawn rate of chest mimics in silver dungeons
added loot to mimics
many mimic loot items cannot be duplicated with skyroot swords

OzPayn committed:
enhancement: gold dungeon loot rebalance

rebalanced gold dungeon loot
rebalanced mimic spawn rates

Burning-Cactus committed:
Attempting a fix to the valkyrie queen despawning

Burning-Cactus committed:
Remove offhand reach modifier

OzPayn committed:
fix: restored accurate 1.2 material block textures in the 1.2 resourc… (#1058)

closes #1053

OzPayn committed:
fix: Valkyrie queens no longer despawn (#1056)

Closes #1054

OzPayn committed:
Dungeon loot rework (#1051)

Full rework of dungeon chest, mimic loot, and boss loot pools
Loot variety is significantly increased as well as having specific item
types be guaranteed spawns for boss loot chests to aid smooth
progression

closes #1040

OzPayn committed:
fix: Remove offhand reach modifier from valkyrie tools (#1059)

Closes #1047

Burning-Cactus committed:
Fix valkyrie lance

OzPayn committed:
updated panorama images

new panorama

OzPayn committed:
Fix: Valkyrie tool attack range (#1061)

Closes #1060

OzPayn committed:
Enhancement: updated panorama images (#1062)

![image](https://user-images.githubusercontent.com/6759864/224515519-e4079d50-1651-48ef-8312-8f80a8155d5f.png)

![image](https://user-images.githubusercontent.com/6759864/224515553-5594725c-bc91-4ef6-9b6b-45e55133a93e.png)

All three dungeons are visible(ish) in the panorama!
Seed - 7595213701795182621
coordinates - -400 123 -55

Might not be final but certainly will be enough for beta

closes #1046

Burning-Cactus committed:
feat: Aerbunnies work the same way as in beta (#1063)

This PR rewrites aerbunnies to act more like they did in old versions,
This includes behavior like fleeing from players and performing
mini-jumps in the air. Additionally, Some of the code is organized into
helper methods to improve readability.

Closes #1044

Burning-Cactus committed:
Improved phoenix armor

Burning-Cactus committed:
Reworked neptune armor

Burning-Cactus committed:
Fixed neptune gloves

Burning-Cactus committed:
Adjust phoenix armor speed

OzPayn committed:
refactor: Reworked phoenix armor and neptune armor set logic (#1065)

Phoenix armor works based off of acceleration instead of the entity's
current speed. Neptune armor uses attributes.

Closes #1064

Burning-Cactus committed:
fix: Sun spirit can now move across the full room (#1057)

Closes #1052

unroman committed:
Updated Ukrainian translation (#1066)

Updated the translation file as i am unable to upload the full
translation file directly to OneSky.
Translated by Nerkin and un_roman

---

I agree to the Contributor License Agreement (CLA).

---------

Co-authored-by: Archie <azwerlein77@gmail.com>
Co-authored-by: Oscar Payn <oscar@gildedgames.com>
Co-authored-by: Burning-Cactus <23108473+Burning-Cactus@users.noreply.github.com>
Co-authored-by: Gilded-Games-Bot <machine@gildedgames.com>

Aether-Team-Bot committed:
Merge branch 'local/baguchan/jei_recipe' into merge/jei_recipe

Aether-Team-Bot committed:
chore: Trigger build

OzPayn reviewed (approved):
lgtm
